daemon/snapper/timestamp_formatting/timestamp_formatting.go

```diff
@@ -0,0 +1,92 @@
+package timestamp_formatting
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type Formatter struct {
+	format   func(time.Time) string
+	location *time.Location
+}
+
+func New(formatString string, locationString string) (*Formatter, error) {
+	location, err := time.LoadLocation(locationString) // no shadow
+	if err != nil {
+		return nil, errors.Wrapf(err, "load location from string %q", locationString)
+	}
+	makeFormatFunc := func(formatString string) (func(time.Time) string, error) {
+		if strings.Contains(formatString, "+") {
```

Comment:
User can enter all sorts of things in the format mask, like "\/|$#", etc. which actually work using format, I don't think guarding against "+" is exactly necessary.

Reply:
Added a comment to explain why we're checking for `+` specifically.

Also, with recent pushes, there's now a config-parsing-time check for all invalid characters: https://github.com/zrepl/zrepl/blob/a8ece782ef0607257b8234dd90fd24f8761b4119/daemon/snapper/snapname/snapname.go#L27-L42